pkg/cluster/deploybaseresources_additional.go

```diff
@@ -20,7 +20,22 @@ import (
 )
 
 func (m *manager) denyAssignment() *arm.Resource {
-	return &arm.Resource{
+	excludePrincipals := []mgmtauthorization.Principal{}
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil && m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil {
```

Comment:
Can use `m.doc.OpenShiftCluster.UsesWorkloadIdentity()`?

Reply:
Yup, thanks for catching this